staging/src/k8s.io/apimachinery/pkg/runtime/scheme.go

```diff
@@ -138,7 +137,6 @@ func (s *Scheme) AddUnversionedTypes(version schema.GroupVersion, types ...Objec
 // the struct becomes the "kind" field when encoding. Version may not be empty - use the
 // APIVersionInternal constant if you have a type that does not have a formal version.
 func (s *Scheme) AddKnownTypes(gv schema.GroupVersion, types ...Object) {
-	s.addObservedVersion(gv)
```

Comment:
I am not sure this is correct.  This line calls `addObservedVersion()` with the GV, the below code calls it for each kind in that GV.  Those are not the same thing.  Without any unit tests, it's hard to say.